### PR TITLE
Allows blind people to touch things to examine them

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Allows blind people to shift-click examine on things in touch range. If they have an empty hand, a do_after will begin, allowing them to examine things if successful.

Also goes through and prevents some looping examines for blind people.

## Why It's Good For The Game

A lot of things are placed in examine text, such as keybinds or information, or if a player is SSD or catatonic. This lets players who are blinded figure out some of these things while also giving them some more interesting flavor (for example, managing to examine a supermatter crystal will dust you).

## Changelog
:cl: Melbert
add: NanoTrasen has added braille support to literally everything. If a blind person has an empty hand, they can now shift-click examine (or right click -> examine) nearby things via touch after a short timer.
/:cl:

### CODE REVIEW

**Comment by MrDoomBringer** [comment]
good Pr. Would be cool if items had separate descriptions for blind people who couldn't necessarily see blinking lights, etc.

unrelated, but I've always wanted to have examining things split into parts, like a purely visual description (This is an access card with wires glued to it), a functional description (These are often used to hack open airlocks, robots, etc), and an OOC/keybind description (Alt+Click it to change this setting or whathaveyou)

**Comment by bluetshirtguy** [comment]
Cool Beans

**Comment by XDTM** [comment]
Does this work on written paper?

**Comment by MrMelbert** [comment]
> Does this work on written paper? 

fun fact: blind people can already read paper, it was fixed, then broken again with new paper - see #51838

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -446,9 +446,31 @@
 		// shift-click catcher may issue examinate() calls for out-of-sight turfs
 		return
 
-	if(is_blind())
-		to_chat(src, "<span class='warning'>Something is there but you can't see it!</span>")
-		return
+	if(is_blind()) //blind people see things differently (through touch)
+		//need to be next to something, awake, on help intent, and have an empty hand
+		if(!in_range(A, src) || incapacitated() || a_intent != INTENT_HELP || LAZYLEN(do_afters) >= get_num_arms() || get_active_held_item())
```
Comment (Jared-Fogle) [inline code/modules/mob/mob.dm:451]: Help intent seems like an unintuitive requirement for this.

File: code/modules/mob/mob.dm [hunk 2]
```diff
@@ -446,9 +446,31 @@
...
+			to_chat(src, "<span class='warning'>Something is there, but you can't see it!</span>")
```
Comment (Jared-Fogle) [inline code/modules/mob/mob.dm:452]: The messaging should probably be different based on *why* you can't examine it. As in, if I examine something with something in my hand, and it doesn't work, it's not obvious to me that I just need to switch to an empty one.

**Comment by Tharcoonvagh** [comment]
>add: NanoTrasen has added braille support to literally everything
>literally everything
>or if a player is SSD or catatonic

File: code/modules/mob/mob.dm (hunk 1 above)
Comment (MrMelbert) [reply code/modules/mob/mob.dm:451]: I used help intent only so you wouldn't punch or shove mobs you were examining. Could be removed, I suppose

Comment (Jared-Fogle) [reply code/modules/mob/mob.dm:451]: I think you should be able to set the intent before attacking, then restore it after.

**Comment by nianjiilical** [comment]
now we just need cockatrice eggs

**Pull request merged**